lib/publiccloud/provider.pm

```diff
@@ -389,9 +388,6 @@ sub create_instances {
             $instance->wait_for_ssh(timeout => $args{timeout},
                 proceed_on_failure => $args{proceed_on_failure}, scan_ssh_host_key => 1);
         }
-
-        $self->upload_boot_diagnostics() if ($args{upload_boot_diagnostics});
```

Comment:
it is called here https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/basetest.pm#L190 second time